services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProviderFactory.java

```diff
@@ -152,6 +158,18 @@ public SAMLIdentityProviderConfig createConfig() {
                             samlIdentityProviderConfig.setEncryptionPublicKey(defaultCertificate);
                         }
                     }
+                    
+                    samlIdentityProviderConfig.setEnabledConfig(entityType.getValidUntil() == null || entityType.getValidUntil().toGregorianCalendar().getTime().after(new Date()) );
+                    
+                	//check for hide on login attribute
+        			if ( entityType.getExtensions() != null && entityType.getExtensions().getEntityAttributes() != null ) {
+        			    for (AttributeType attribute :  entityType.getExtensions().getEntityAttributes().getAttribute()) {
+        				    if (MACEDIR_ENTITY_CATEGORY.equals(attribute.getName()) && attribute.getAttributeValue().contains(REFEDS_HIDE_FROM_DISCOVERY) )
```

Comment:
Prevent potential NPE
```suggestion
        				    if (MACEDIR_ENTITY_CATEGORY.equals(attribute.getName()) && REFEDS_HIDE_FROM_DISCOVERY.equals(attribute.getAttributeValue()))
```

Reply:
attributeValue is a List in class AttributeType which is instantiated as empty list as you can see.
Moreover it is error to check equality between String and List.

Reply:
You're indeed right :-)